### PR TITLE
ENH add ability to use relu to autoencoder

### LINKED ISSUE #21: enable relu activation for autoencoder

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the ability to use relu activations for the autoencoder.

Changes in the tests include 
1. now use `relu` for the activation
2. renamed a bunch of them
3. changed the size of the hidden layers from 1 unit to 2 in many cases
4. made sure learning rates were set properly to get good results

Closes #21

### CODE REVIEW

File: muffnn/autoencoder/autoencoder.py [hunk 1]
```diff
@@ -43,7 +43,10 @@ class Autoencoder(TFPicklingBase, TransformerMixin, BaseEstimator):
         dropout will not be used. cf. `TensorFlow documentation
         <https://www.tensorflow.org/versions/r0.11/api_docs/python/nn.html#dropout>`
     hidden_activation : tensorflow graph operation, optional
-        The activation function for the hidden layers and encoding layer.
+        The activation function for the hidden layers.
+        See `tensorflow.nn` for various options.
+    encoding_activation : tensorflow graph operation, optional
```
Comment (mheilman) [inline muffnn/autoencoder/autoencoder.py]: I think it would probably help to say that `None` means linear activation.

File: muffnn/autoencoder/tests/test_autoencoder.py [hunk 2]
```diff
@@ -23,6 +23,8 @@
 _LOGGER = logging.getLogger(__name__)
 iris = load_iris()
 
+LEARNING_RATE = 2e-3
```
Comment (mheilman) [inline muffnn/autoencoder/tests/test_autoencoder.py]: Why is this needed here?  It might be worth adding a comment.

File: muffnn/autoencoder/tests/test_autoencoder.py [hunk 3]
```diff
@@ -505,48 +568,49 @@ def test_sparse_inputs():
     for sparse_type in [None, 'csr', 'bsr', 'coo', 'csc', 'dok', 'lil']:
         scores.append(_check_ae(
             5.0,
+            hidden_units=(2,),
             sparse_type=sparse_type,
             cat_inds=range(4),
-            n_epochs=1000))
+            n_epochs=1000,
+            learning_rate=5e-3))
```
Comment (mheilman) [inline muffnn/autoencoder/tests/test_autoencoder.py]: Why not `LEARNING_RATE`?  Maybe that should be `DEFAULT_LEARNING_RATE`?

Comment (beckermr) [inline muffnn/autoencoder/tests/test_autoencoder.py]: I upped the learning rate here to make sure the autoencoder converged. Not needed for the tests per se, but def useful to make sure it could. I will leave a comment.

File: muffnn/autoencoder/tests/test_autoencoder.py (hunk 2 above)
Comment (beckermr) [inline muffnn/autoencoder/tests/test_autoencoder.py]: This is a convenience thing. I will leave a note.

**Comment by beckermr** [comment]
I left comments on the tests at the top, @mheilman!

**Comment by mheilman** [comment]
LGTM
